source/common/router/config_impl.cc

```diff
@@ -752,12 +752,7 @@ RouteEntryImplBase::RouteEntryImplBase(const CommonVirtualHostSharedPtr& vhost,
                                Http::Headers::get().MethodValues.Connect) ||
         absl::EqualsIgnoreCase(upgrade_config.upgrade_type(),
                                Http::Headers::get().UpgradeValues.ConnectUdp)) {
-      if (Runtime::runtimeFeatureEnabled(
-              "envoy.reloadable_features.http_route_connect_proxy_by_default")) {
-        if (upgrade_config.has_connect_config()) {
-          connect_config_ = std::make_unique<ConnectConfig>(upgrade_config.connect_config());
-        }
-      } else {
+      if (upgrade_config.has_connect_config()) {
         connect_config_ = std::make_unique<ConnectConfig>(upgrade_config.connect_config());
       }
     } else if (upgrade_config.has_connect_config()) {
```

Comment:
Something seems wrong with `else if` condition. Isn't it the same as the `if` condition above?

Reply:
I believe it's correct but I agree it could be clearer. Let's put the common condition into a parent `if` instead, if that sounds good to you.